### PR TITLE
Add docs for static files

### PR DESCRIPTION
**Pull request by erikzaadi**

### CODE REVIEW

File: docs/user_guide.md [hunk 1]
```diff
@@ -379,6 +379,26 @@ The following types are supported in the language:
...
+
+This simplifies viewing and editing those files in separate instead of a huge string block inside a NaCl file.
+
+```hcl
+salesforce.Text ApexFileForProfile {
+    content = file("salesforce/classes/ApexFileForProfile.cls")
+}
+```
+
+Those files are put in the `static-resources` folder inside your environment.
```
Comment (adamro) [inline docs/user_guide.md:394]: shouldn't `environment` should be `workspace`?

File: docs/user_guide.md [hunk 2]
```diff
@@ -379,6 +379,26 @@ The following types are supported in the language:
 
 ## Glossary - Coming Soon!
 
+### Static Files
+
+Static files are a function (will be elaborated in the spec) which references a file external to the NaCl file.
```
Comment (gilhoffer) [inline docs/user_guide.md:384]: This is a very technical view, I don't think the users at this point should really care that it's implemented as a function, rather I would focus on why and what.

File: docs/user_guide.md [hunk 3]
```diff
@@ -379,6 +379,26 @@ The following types are supported in the language:
...
+
+This simplifies viewing and editing those files in separate instead of a huge string block inside a NaCl file.
```
Comment (gilhoffer) [inline docs/user_guide.md:386]: missing "files" after separate?

File: docs/user_guide.md (hunk 1 above)
Comment (gilhoffer) [inline docs/user_guide.md:394]: put --> stored ?
As the multi-env behavior is non-standard, we might need to add a short section also in the multi-env section below.

File: docs/user_guide.md [hunk 4]
```diff
@@ -379,6 +379,26 @@ The following types are supported in the language:
...
+salesforce.Text ApexFileForProfile {
+    content = file("salesforce/classes/ApexFileForProfile.cls")
+}
+```
+
+Those files are put in the `static-resources` folder inside your environment.
+
+The files are validated and hashed (MD5) for comparison to see if there are any changes.
+
+If the `file` function points to a non existing file, the preview / deploy operations will warn and stop.
```
Comment (gilhoffer) [inline docs/user_guide.md:398]: - Should we add something on default folder naming?
- Should we add something on rename/remove behavior?

File: docs/user_guide.md [hunk 5]
```diff
@@ -388,7 +408,6 @@ The following types are supported in the language:
 - Annotation
 - Type
 - Settings
-- State file(s)
```
[the remark this answers is not in the document]
Comment (erikzaadi) [reply docs/user_guide.md]: Mistake, updated

**Comment by erikzaadi** [comment]
@gilhoffer @adamro please look again at the updated version

File: docs/user_guide.md [hunk 6]
```diff
@@ -326,6 +357,7 @@ For example, a workspace with 3 environments (named dev, test and prod), each co
     — dev/
         — salesforce/
         — hubspot/
+        — static-resources/
```
Comment (gilhoffer) [inline docs/user_guide.md:360]: Did you add the `static-resources` dir only to the dev env on-purpose?

Comment (erikzaadi) [reply docs/user_guide.md:360]: yeah to show that it's not mandatory per env folder

Comment (gilhoffer) [reply docs/user_guide.md:360]: cool then :)

**Pull request merged**